portal-api/src/drivers/fileshare.js

```diff
@@ -1,20 +1,31 @@
 const { ShareServiceClient, StorageSharedKeyCredential } = require('@azure/storage-file-share');
 const fetch = require('node-fetch');
+const { FILESHARES } = require('../constants');
 
-const { AZURE_WORKFLOW_FILESHARE_CONFIG, AZURE_PORTAL_FILESHARE_CONFIG } = require('../config/fileshare.config');
+const { AZURE_WORKFLOW_FILESHARE_CONFIG, AZURE_PORTAL_FILESHARE_CONFIG, AZURE_UTILISATION_REPORTS_FILESHARE_CONFIG } = require('../config/fileshare.config');
 
 let userDefinedConfig;
 
 const setConfig = (fileshareConfig) => {
   userDefinedConfig = fileshareConfig;
 };
 
-const getConfig = (fileshare = 'portal') => {
-  const config = fileshare === 'workflow' ? AZURE_WORKFLOW_FILESHARE_CONFIG : AZURE_PORTAL_FILESHARE_CONFIG;
+const getConfig = (fileshare = FILESHARES.PORTAL) => {
+  let config;
+  switch (fileshare) {
+    case FILESHARES.WORKFLOW:
+      config = AZURE_WORKFLOW_FILESHARE_CONFIG;
+      break;
+    case FILESHARES.UTILISATION_REPORTS:
+      config = AZURE_UTILISATION_REPORTS_FILESHARE_CONFIG;
+      break;
+    default:
```

Comment:
I'd probably add `case FILESHARES.PORTAL` here as well and let it ffall through to the default blow just to be clear it exists as an option

Reply:
Do we ever expect this default case to be used? If not then maybe we should throw an error instead as this could lead to unexpected results.

Reply:
Just realised I think we need to return the `userDefinedConfig` if defined, only reverting to env var defined config if not defined.